src/session-factory.ts

```diff
@@ -165,7 +198,7 @@ export class SessionFactory
    * @throws {Error} If the session is invalid or cannot be released.
    */
   release(session: Session): void {
-    if (!this.isMultiplexed) {
+    if (!session.metadata.multiplexed) {
```

Comment:
do we need optional check for "metadata" here ?

Reply:
made it optional